src/api/spec/controllers/webui/request_controller_spec.rb

```diff
@@ -13,20 +13,20 @@
   let(:target_package) { create(:package_with_file, name: 'goal', project_id: target_project.id) }
   let(:source_project) { submitter.home_project }
   let(:source_project_fluffy) { submitter_with_group.home_project }
-  let(:source_package) { create(:package, name: 'ball', project_id: source_project.id) }
+  let(:source_package) { create(:package, :as_submission_source, name: 'ball', project: source_project) }
   let(:devel_project) { create(:project, name: 'devel:project') }
   let(:devel_package) { create(:package_with_file, name: 'goal', project: devel_project) }
-  let(:bs_request) { create(:bs_request, description: 'Please take this', creator: submitter.login) }
-  let(:bs_request_submit_action) do
-    create(:bs_request_action_submit, target_project: target_project.name,
-                                      target_package: target_package.name,
-                                      source_project: source_project.name,
-                                      source_package: source_package.name,
-                                      bs_request_id: bs_request.id)
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           description: 'Please take this', creator: submitter.login,
```

Comment:
[nitpick] `.login` is not needed (and in the other 100 places where this happens).

Reply:
See https://github.com/openSUSE/open-build-service/pull/6721 :)